### ISSUE TITLE
False positive in SA1119, if (!(value is Type type)) erroneously reports inner parentheses unnecessary.

### ISSUE THREAD

**Issue by Flynn1179** [DotNetAnalyzers/StyleCopAnalyzers#2517]
This is a new one in C#7

The ability to also declare a variable using a type pattern causes the unnecessary parenthesis warning to appear.

This works, and correctly allows it with no warning:

```csharp
if (!(value is Type)) { .. }
```

This doesn't:

```csharp
if (!(value is Type type)) { /* do something with type */ }
```

The inner parenthesis are reported as unnecessary, but they're definitely not.

**Comment by sharwell** [comment]
Duplicate of #2372 :smile:

**Comment by Flynn1179** [comment]
Odd, I searched for SA1119, didn't find it. Sorry!

**Comment by sharwell** [comment]
@Flynn1179 No problem, we prefer a duplicate than no issue filed at all :smile:

**Comment by Flynn1179** [comment]
Ah, identified my error: I left the default is:open clause in the search, and didn't consider the possibility it had already been fixed, having updated to the latest released version this morning. Lesson learned!
